src/HealthChecks.Redis/RedisHealthCheck.cs

```diff
@@ -9,14 +9,19 @@ namespace HealthChecks.Redis;
 /// </summary>
 public class RedisHealthCheck : IHealthCheck
 {
-    private static readonly ConcurrentDictionary<string, IConnectionMultiplexer> _connections = new();
-    private readonly string? _redisConnectionString;
+    private static readonly ConcurrentDictionary<Func<CancellationToken, Task<string?>>, IConnectionMultiplexer> _connections = new();

```

Comment:
Is this caching now broken if I create multiple `RedisHealthCheck` instances with the same connection string value? They each will get a different delegate instance now, where before it was keyed off the actual string value.